runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/ServerFactory.java

```diff
@@ -144,7 +145,9 @@ private static Server createServer(List<BindableService> services, InetSocketAdd
           NettyServerBuilder.forPort(socket.getPort())
               // Set the message size to max value here. The actual size is governed by the
               // buffer size in the layers above.
-              .maxMessageSize(Integer.MAX_VALUE);
+              .maxMessageSize(Integer.MAX_VALUE)
+              .permitKeepAliveTime(1, TimeUnit.SECONDS)
+              .permitKeepAliveWithoutCalls(true);
```

Comment:
It seems to me, we're doing something wrong in the client configuration. This is ok as a workaround if this is a bug, but perhaps we should check the client configuration?

Reply:
We are not checking the client configuration anywhere. 
However, I also feel that this might be a server/client configuration issues.
This is just a work around till we find the root cause.

Reply:
What if we set the keep alive time explicitly on both ends? Just wondering whether it made sense to set the values instead of using this workaround.

https://github.com/grpc/grpc/blob/master/doc/keepalive.md#defaults-values

Reply:
I will try it and see if it works. 
The root cause is still unknown so I am not sure setting explicit config is a fix or just another work around.

Reply:
Let me know if that is feasible. Otherwise let's merge the PR.

Reply:
If the runs were successful could you push the changes here?

Reply:
Updated the PR to set 20sec keep alive time in python and Java.